overlord/state/notices.go

```diff
@@ -107,15 +119,83 @@ func flattenUserID(userID *uint32) (uid uint32, isSet bool) {
 	return *userID, true
 }
 
+// Type represents a group of notices originating from a common source.
```

Comment:
At this point, can we just make all these fields (except the new `repeatCheckData` field) exported, i.e., capitalized and remove these functions? (with `noticeType` renamed `Type`) That's more common in Go, and I don't think there's any real advantage of private fields plus exported getters.

Reply:
Updated, Thanks!